File: Recursion/Lecture 01/CodeStudio/Factorial of a Number.py
# Your code goes here
# this function fac() will return us the factorial of the number n
def fac(n):
    if n == 0:
        return 1
        
    # recursion call
    return n*fac(n-1)  


n = int(input())
if n < 0:
    print("Error")
else:
    print(fac(n))


""" 
    time complexity: O(n)
    space complexity: O(n) 
"""

File: Recursion/Lecture 01/CodeStudio/Factorial of a Number.java
// This Factorial only works small numbers like 1-15
import java.io.*;
import java.util.*;

class Solution {

    // this function fac() will return us the factorial of the number n
	static int fac(int n){
		if(n==0) return 1;
		
		// recursion call
		return n*fac(n-1); 
	}

	public static void main(String args[]) {
		// Write code here
		Scanner sc=new Scanner(System.in);
		int n=sc.nextInt();
		if(n<0){
			System.out.println("Error");
		}
		else{
			System.out.println(fac(n));
		}
	}
}

/*
    time complexity : O(n)
    space complexity : O(n)
*/
